artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -847,6 +847,7 @@ public MessageWriter selectOutgoingMessageWriter(ProtonServerSenderContext sende
             if (message.isLargeMessage()) {
                selected = largeMessageWriter != null ? largeMessageWriter :
                   (largeMessageWriter = new AMQPLargeMessageWriter(sender));
+               largeMessageWriter.openContext(reference);
```

Comment:
thing is.. it's always asynchronous... except to large messages... (since the previous interaction already you had the current large message returning something to make the handler return BUSY).

So.. in all the other cases it should be treating through the acceptor.. except to this large message case here.

Reply:
So are you saying a handler can now accept an additional message while it's trying to process the previously passed large message ?  That seem rife with errors. Otherwise I don't follow what you are saying.

Reply:
this method only makes sense in the context of the AMQP Large Message. It doesn't make sense in any other writers (not yet at least).  

I've tried to make them all use this method, but the other writers were writing it asynchronously as I mentioned earlier.

Reply:
I've made the change here locally and I don't see any reason it doesn't work to just make the open method accept the message reference and do the initialization there as opposed to adding a second "special" open method the select of the writer is done in the same thread as the sender will be calling open prior to handling the write into the connection thread.

Reply:
I made the change.

I had done some back and forth changes and I couldn't see it now.. thanks for the review.